test/support/factories/detour_factory.ex

```diff
@@ -73,6 +75,43 @@ defmodule Skate.DetourFactory do
         put_in(state["value"], %{"Detour Drawing" => "Past"})
       end
 
+      def with_route(%Skate.Detours.Db.Detour{} = detour, %{name: _, id: _} = route) do
+        %{detour | state: with_route(detour.state, route)}
+      end
+
+      def with_route(
+            %{"context" => %{"route" => %{"directionNames" => direction_names}}} = state,
+            %{name: route_name, id: route_id}
+          ) do
+        put_in(state["context"]["route"], %{
+          "id" => route_id,
+          "name" => route_name,
+          "directionNames" => direction_names
+        })
+      end
```

Comment:
nit(non-blocking): `with_route_name` and `with_route_id` are unused here, I'd prefer to invoke them separately when needed, but you may consider this to avoid needing to populate the other keys present, like `direction_names`.
```suggestion
      def with_route(%Skate.Detours.Db.Detour{} = detour, %{name: _, id: _} = route) do
        %{detour | state: with_route(detour.state, route)}
      end

      def with_route(
            %{"context" => %{"route" => %{}}} = state,
            %{name: route_name, id: route_id}
          ) do
        state
        |> with_route_id(route_id)
        |> with_route_name(route_name)
      end
```